src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -0,0 +1,102 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "BlockLogicalCoordinates.hpp"
+
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/Domain.hpp"  // IWYU pragma: keep
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Literals.hpp"
+
+template <size_t Dim>
+void BlockLogicalCoordHolder<Dim>::pup(PUP::er& p) noexcept {
+  p | block_id;
+  p | block_logical_coords;
+}
+
+template <size_t Dim>
+bool operator==(const BlockLogicalCoordHolder<Dim>& lhs,
+                const BlockLogicalCoordHolder<Dim>& rhs) noexcept {
+  return lhs.block_id == rhs.block_id and
+         lhs.block_logical_coords == rhs.block_logical_coords;
+}
+template <size_t Dim>
+bool operator!=(const BlockLogicalCoordHolder<Dim>& lhs,
+                const BlockLogicalCoordHolder<Dim>& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+template <size_t Dim, typename Frame>
+std::vector<BlockLogicalCoordHolder<Dim>> block_logical_coordinates(
+    const Domain<Dim, Frame>& domain,
+    const tnsr::I<DataVector, Dim, Frame>& x) noexcept {
+  const size_t num_pts = get<0>(x).size();
+  std::vector<BlockLogicalCoordHolder<Dim>> block_logical_coords(num_pts);
```

Comment:
This name is a bit confusing since block_logical_coords is also a member variable of the struct, how about
`logical_coords_of_all_blocks`?

Reply:
Its not the logical coords of the blocks, its the logical coords of the points.  `block_logical_coords_of_all_points` seems too long...